medusa/storage/s3_rgw.py

```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Spotify AB
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import io
+import json
+import logging
+import os
+
+from dateutil import parser
+from libcloud.storage.drivers.rgw import S3RGWStorageDriver
+
+from medusa.storage.abstract_storage import AbstractStorage
+
+
+class S3RGWStorage(AbstractStorage):
+
+    def connect_storage(self):
+        with io.open(os.path.expanduser(self.config.key_file), 'r', encoding='utf-8') as json_fi:
+            credentials = json.load(json_fi)
+
+        driver = S3RGWStorageDriver(
+            host=self.config.host,
+            port=self.config.port,
+            region="default",
+            signature_version="4",
+            key=credentials['aws_access_key_id'],
```

Comment:
It seems a bit confusing to use a credential file that mentions AWS although we're not using an AWS account (unless I'm missing something).
Is it a standard way of authenticating to CEPH using the S3 api? Are aws credentials involved in any way?

Reply:
Good point, they strive to keep the nomenclature the same but I clearly forgot to remove the `aws_`. Done.